classes/class-apm-helper.php

```diff
@@ -0,0 +1,225 @@
+				}
+				switch ( $type ) {
+					case 'text':
+						$default_value = ( isset( $settings['def_value'] ) ) ? $settings['def_value'] : '';
+						$input .= '<h4>' . $label . '</h4>';
+
+						if ( isset( $settings['help'] ) && '' != $settings['help'] ) {
+							$input .= '<p class="admin-help">' . $settings['help'] . '</p>';
+						}
+
+						$input .= '<input type="text" name="' . $id . '" id="' . $id . '" class="regular-text ' . $class . '" placeholder="' . $placeholder . '" value="' . $default_value . '"/>';
+						break;
+
+					case 'hidden':
+						$default_value = ( isset( $settings['def_value'] ) ) ? $settings['def_value'] : '';
+						$input .= '<input type="hidden" name="' . $id . '" id="' . $id . '" class="' . $class . '" value="' . $default_value . '"/>';
+						break;
+
+					case 'submit':
+						$default_value = ( isset( $settings['def_value'] ) ) ? $settings['def_value'] : '';
+
+						if ( isset( $settings['help'] ) && '' != $settings['help'] ) {
+							$input .= '<p class="admin-help">' . $settings['help'] . '</p>';
+						}
+
+						$input .= '<p class="submit"><input type="submit" name="' . $id . '" id="' . $id . '" class="button-primary ' . $class . '" value="' . $default_value . '"/>';
+						if ( isset( $settings['spinner'] ) && $settings['spinner'] ) {
+							$input .= '<span class="spinner ap_mautic_spinner" style=""></span></p>';
+						}
+						if ( isset( $settings['nonce_acion'] ) && '' != $settings['nonce_acion'] ) {
+							$input .= wp_nonce_field( $settings['nonce_acion'], $settings['nonce_name'] );
+						}
+						break;
+
+					case 'button':
+						$default_value = ( isset( $settings['def_value'] ) ) ? $settings['def_value'] : '';
+
+						if ( isset( $settings['help'] ) && '' != $settings['help'] ) {
+							$input .= '<p class="admin-help">' . $settings['help'] . '</p>';
+						}
+						$input .= '<p class="submit"><input type="button" name="' . $id . '" id="' . $id . '" class="button-primary ' . $class . '" value="' . $default_value . '"/></p>';
+						if ( isset( $settings['nonce_acion'] ) && '' != $settings['nonce_acion'] ) {
+							$input .= wp_nonce_field( $settings['nonce_acion'], $settings['nonce_name'] );
+						}
+					break;
+
+					case 'checkbox':
+						$checked = ( isset( $settings['ischecked'] ) ) ? $settings['ischecked'] : '';
+						$input .= '<h4>' . $label . '</h4>';
+						if ( isset( $settings['help'] ) && '' != $settings['help'] ) {
+							$input .= '<p class="admin-help">' . $settings['help'] . '</p>';
+						}
+						$input .= '<input type="checkbox" name="' . $id . '" id="' . $id . '" class="' . $class . '" value="" "' . checked( 1, $checked, false ) . '"/>' . $settings['text'];
+					break;
+
+					default:
+						$input .= '';
+						break;
+				}
+
+				if ( isset( $settings['desc'] ) && '' != $settings['desc'] ) {
+					$input .= '<p class="admin-help admin-field-desc">' . $settings['desc'] . '</p>';
+				}
+				if ( $iswrap ) {
+					$input .= '</div>';
+				}
+			}
+
```

Comment:
`render_input_html()` has many instances that require correct escaping, I have not pinpointed each and every instance, please have a look at the whole function.